codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Exception.java

```diff
@@ -47,6 +63,13 @@ public Http2Error error() {
     }
 
     /**
+     * If a connection error should attempt a graceful shutdown.
+     * @return {@code true} if a connection error should attempt a graceful shutdown. {@code false} otherwise.
+     */
+    public boolean gracefulShutdown() {
```

Comment:
Actually I think that having both this and `isStreamError` is a bit confusing.  IMO this discussion is orthogonal to our exception hierarchy .. I don't think we need to change any of that. I actually think that having an enum like this (maybe `ShutdownHint` would be a better name?) could make our shutdown decision making more straight forward. How hard do you think it would be to change the `isStreamError` handling logic over to this?